Update dogm-graphic.h

7h0mas-R opened the pull request:
The command in line 444 should be the same as in line 441 and according to EA datasheet and UC1701 datasheet the last four (not 5) bits are the page-address. 
It probably only rarely becomes an issue, if users never target an address > 15
The command for setting temperature compensation mode is [TC, 0, 0, 1], so it should be 0x90 for high and 0x10 for low (also stated in both datasheets, so I assume correct).

Did not have the chance to confirm on hardware yet, whether the mistake is in the code or the datasheets, but at least the first is either incorrect on line 444 or 441. Thought I better post it now, before I forget.

7h0mas-R committed:
Update dogm-graphic.h

The command in line 444 should be the same as in line 441 and according to EA datasheet and UC1701 datasheet the last four (not 5) bits are the page-address. 
It probably only rarely becomes an issue, if users never target an address > 15
The command for setting temperature compensation mode is [TC, 0, 0, 1], so it should be 0x90 for high and 0x10 for low (also stated in both datasheets, so I assume correct).

Did not have the chance to confirm on hardware yet, whether the mistake is in the code or the datasheets, but at least the first is either incorrect on line 444 or 441. Thought I better post it now, before I forget.

mueschel commented:
Thanks for this fix, the number of pages is for sure wrong.


But LCD_TEMPCOMP_HIGH = 0x80 was correct. There is just this one bit to select the temperature compensation. The 0x10 bit is set when calling the actual set function that is defined in line 450:

```
#define LCD_SET_ADV_PROG_CTRL(i)     lcd_command(LCD_ADV_PROG_CTRL); \
                                       lcd_command(LCD_ADV_PROG_CTRL2 | i)
```
because LCD_ADV_PROG_CTRL2 == 0x10

7h0mas-R committed:
Update dogm-graphic.h

right... only the address mask is wrong.
must have been too tiered to oversee the masking of the temp compensation...

7h0mas-R commented:
you are right, did not see it at first glance... 
reverted the change in patch-1
